airflow/providers/databricks/hooks/databricks.py

```diff
@@ -656,14 +655,20 @@ def get_repo_by_path(self, path: str) -> str | None:
 
         return None
 
-    def update_job_permission(self, json: dict[str, Any]) -> dict:
+    def update_job_permission(self, job_id: int, json: dict[str, Any]) -> dict:
         """
         Update databricks job permission.
 
+        :param job_id: job id
         :param json: payload
         :return: json containing permission specification
         """
-        return self._do_api_call(UPDATE_PERMISSION_ENDPOINT, json)
+        response = {}
+        try:
+            response = self._do_api_call(("PATCH", f"api/2.0/permissions/jobs/{job_id}"), json)
+        except Exception as e:
+            self.log.error("Could not update permission %s", e)
```

Comment:
IMO , if user is passing ACL and is expecting the permissions to be updated, then we should reraise the exception, otherwise the task will be marked as success without the expected outcome of permissions being updated too. WDYT?

Reply:
Agree with @pankajkoti  👍

Reply:
Removed try catch block. This will throw error if there is an error in updating job permission.